clients/src/main/java/org/apache/kafka/clients/ClientResponse.java

```diff
@@ -104,6 +135,50 @@ public long requestLatencyMs() {
         return latencyMs;
     }
 
+    public Logger getLogger() {
+        if (logContext == null) {
+            LogContext logContext = new LogContext("[" + requestHeader.toString() + "] ");
+            return logContext.logger(ClientResponse.class);
+        }
+        return logContext.logger(ClientResponse.class);
+    }
+
+    public void releaseBuffer() {
+        if (memoryPool != null && responsePayload != null) {
+            getLogger().trace(
+                "ByteBuffer[{}] returned to memorypool ({}). Ref Count: {}. RequestType: {}",
+                (responsePayload == null ? "null" : responsePayload.position()), (memoryPool == null ? "null" : "not null"),
+                refCount.get(), this.requestHeader.apiKey());
+            memoryPool.release(responsePayload);
+            responsePayload = null;
+        } else {
+            // Mostly this means double releaseBuffer calls or releaseBuffer where response is null such as disconnects
+            getLogger().error(
+                "ByteBuffer[{}] returned to memorypool ({}). Ref Count: {}. RequestType: {}",
+                (responsePayload == null ? "null" : responsePayload.position()), (memoryPool == null ? "null" : "not null"),
+                refCount.get(), this.requestHeader.apiKey());
+        }
+    }
+
+    public void checkBufferRelease() {
```

Comment:
protected ?

Reply:
Should we move this to ClientResponseWithFinalize class since this has to be specifically enabled using a config?
That way we don't modify the existing class and its behavior.
Thoughts??

Reply:
This method seems to be not just checking but also releasing the buffer. Should it be named to better indicate what it is doing?

Reply:
Renamed to `checkAndForceBufferRelease()` and moved to ClientResponseWithFinalize.java.

Reply:
Also, just for clarification. The method is more like a debugging tool to make sure all the buffers allocated are accounted for and forcefully releasing the buffers to prevent memory pool running out of buffers to handout.